src/DotLiquid/Context.cs

```diff
@@ -716,7 +704,11 @@ private static bool IsKeyValuePair(object obj)
             if (obj != null)
             {
                 Type valueType = obj.GetType();

```

Comment:
Null check is superfluous (line 650), as is GetType() call which is now assigned to objType on line 667.